packages/frontend/src/components/nft/NFTDetail.js

```diff
@@ -0,0 +1,196 @@
+            drop-shadow(0px 41.7776px 33.4221px rgba(0, 0, 0, 0.0503198)) 
+            drop-shadow(0px 22.3363px 17.869px rgba(0, 0, 0, 0.0417275)) 
+            drop-shadow(0px 12.5216px 10.0172px rgba(0, 0, 0, 0.035)) 
+            drop-shadow(0px 6.6501px 5.32008px rgba(0, 0, 0, 0.0282725)) 
+            drop-shadow(0px 2.76726px 2.21381px rgba(0, 0, 0, 0.0196802));
+        }
+
+        .desc {
+            margin-top: 48px;
+            margin-bottom: 56px;
+
+            font-weight: 500;
+            font-size: 16px;
+            line-height: 150%;
+            display: flex;
+            align-items: center;
+            color: #272729;
+        }
+    }
+
+    .owner {
+        p {
+            font-weight: 500;
+            font-size: 12px;
+            line-height: 150%;
+            display: flex;
+            align-items: center;
+            letter-spacing: 0.115em;
+            color: #A2A2A8;
+        }
+
+        .inner {
+            height: 30px;          
+            line-height: 30px; 
+            display: flex;         
+            align-items: center;
+
+            span {
+                font-weight: 500;
+                font-size: 16px;
+                line-height: 150%;
+                display: flex;
+                align-items: center;
+                color: #272729;
+            }
+        }
+    }
+
+    &&& {
+        .transfer-btn {
+            width: 100%;
+            margin-top: 56px;
+
+            svg {
+                margin-right: 10px;
+            }
+        }
+    }
+
+    .back-arrow-button {
+        position: absolute;
+        top: -12px;
+        left: -78px;
+    }
+
+`;
+
+const UserIcon = styled.div`
+    background-size: 21px;
+    flex: 0 0 40px;
+    height: 40px;
+    border-radius: 50%;
+    background-color: #f8f8f8;
+    text-align: center;
+    margin: 0 12px 0 0;
+    
+    svg {
+        width: 26px;
+        height: 26px;
+        margin: 7px;
+    }
+
+    @media (min-width: 940px) {
+        display: inline-block;
+    }
+`;
+
+function transferSVG() {
+    return (
+        <svg className="transfer-svg" width="22" height="22" viewBox="0 0 22 22" fill="none" xmlns="http://www.w3.org/2000/svg">
+          <path d="M21 1L14 21L10 12L1 8L21 1Z" stroke="#A2A2A8" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
+          <path d="M21 1L10 12" stroke="#A2A2A8" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round"/>
+        </svg>
+    );
+}
+
+export function NFTDetail({ match, location, history }) {
+    const accountId = useSelector(selectAccountId);
+    const fungibleTokens = useFungibleTokensIncludingNEAR();
+    const nearBalance = fungibleTokens[0].balance;
+
+    const contractId = match.params.contractId;
+    const tokenId = match.params.tokenId;
+    const [ nft, setNft ] = useState();
+    const [ ownerId, setOwnerId ] = useState();
+    const [transferNftDetail, setTransferNftDetail] = useState();
```

Comment:
You can use the `contractId` and `tokenId` from the url param and check if the NFT is part of the users list of NFTs so that you don't have to make the call below and then set with `setOwnerId`. 

You can use the `const sortedNFTs = useSelector((state) => selectTokensWithMetadataForAccountId(state, { accountId }));` used in `Wallet.js`

Reply:
That's a good point. I suggest that we check if there's a token list present when the user lands on this page with `useEffect`, and if there's none, dispatch `fetchNFTs` to get the list. It's better if we use one source of truth for the users NFTs to avoid potential issues.

Reply:
hmm actually I don't think this is a good option for two reasons:
1. the URL could be opened by anybody, not just the NFT owner, so his NFT list doesn't necessarily have this one
2. load the entire NFT list costs too much in order to display one single NFT

how do you think